client/src/components/Patients/index.js

```diff
@@ -1,32 +1,64 @@
 import React, { useState, useEffect } from 'react';
-import { Table, Typography, Input, message, Spin } from 'antd';
+import { Table, Typography, Input, message, Spin, Select } from 'antd';
 import axios from 'axios';
 import './style.css';
 import { useHistory } from 'react-router-dom';
+import { differenceInCalendarYears, parse } from 'date-fns';
 
 const { Search } = Input;
 const { Title } = Typography;
+const { Option } = Select;
 
 const Patients = () => {
   const [dataSource, setDataSource] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [searchKey, setSearchKey] = useState('firstName');
   const history = useHistory();
 
-  const fetchPatients = async () => {
+  const buildUrlQuery = (params) => {
+    const usp = new URLSearchParams();
+    const newParams = params || {};
+    Object.keys(newParams).forEach((key) => {
+      const value = newParams[key];
+      if (value) usp.set(key, value);
+    });
+    return usp.toString();
+  };
+
+  const fetchPatients = async (searchQuery) => {
     try {
+      setLoading(true);
+      if (typeof searchQuery === 'string' && searchQuery.trim().length) {
+        setLoading(true);
+        const search = {};
+        search[searchKey] = searchQuery;
+        const {
+          data: { data },
+        } = await axios.get(`/api/v1/patients/search?${buildUrlQuery(search)}`);
+        setDataSource(data);
+        setLoading(false);
+        return data;
+      }
       setLoading(true);
       const {
         data: { data },
       } = await axios.get(`/api/v1/patients`);
-      setDataSource(data);
       setLoading(false);
+      setDataSource(data);
       return data;
     } catch {
       setLoading(false);
       return message.error('Sever error, please try again later');
     }
   };
 
+  const onSearch = async (value) => {
+    setLoading(true);
+    const patients = await fetchPatients(value);
+    setDataSource(patients);
+    setLoading(false);
+  };
+
   useEffect(() => {
```

Comment:
I think you don't need to use - useEffect for search 
It just a request make by a function - onClick or onPressKey and when data came you can render it.

we did a similar of this component [here](https://github.com/GSG-G9/dentoro/blob/fca79d780b7a00f3743a3e0dace047375a3078c2/client/src/components/CalendarSearch/index.js#L8) please view it and do your best.